pkg/gofr/http/responder.go

```diff
@@ -58,6 +63,45 @@ func (r Responder) Respond(data any, err error) {
 	_ = json.NewEncoder(r.w).Encode(resp)
 }
 
+func (r Responder) determineResponse(data any, err error) (statusCode int, errObj any) {
+	// Handle empty struct case first
+	if err != nil && isEmptyStruct(data) {
+		return http.StatusInternalServerError, createErrorResponse(errEmptyResponse)
+	}
+
+	statusCode, errorObj := getStatusCode(r.method, data, err)
+
+	return statusCode, errorObj
```

Comment:
```suggestion

	statusCode, errorObj := getStatusCode(r.method, data, err)
	
	if statusCode == 0 {
		statusCode = http.StatusInternalServerError
	}

	return statusCode, errorObj
```

We need to check for statusCode being 0, in which case WriteHeader will panic.

Reply:
Sure corrected